cpp/src/slice2cs/Gen.cpp

```diff
@@ -2111,18 +2111,12 @@ Slice::Gen::TypesVisitor::visitStructStart(const StructPtr& p)
 
     emitAttributes(p);
     emitPartialTypeAttributes();
-    _out << nl << "public " << (classMapping ? "sealed partial class" : "partial record struct") << ' ' << name;
-
-    StringList baseNames;
-
-    if (classMapping)
-    {
-        baseNames.push_back("System.IEquatable<" + name + ">");
-    }
+    _out << nl << "public " << (classMapping ? "sealed partial record class" : "partial record struct") << ' ' << name;
```

Comment:
It is common for record classes and structs to be read-only. However, here they are read-write for backwards source compatibility.